### PR TITLE
build(gha): Change frontend workflows to always run

### PR DESCRIPTION
**Pull request by billyvg**
This changes the frontend workflows to always run (similar to what we did with the backend path filters).

### CODE REVIEW

File: .github/file-filters.yml [hunk 1]
```diff
@@ -12,12 +12,12 @@ frontend:
   - 'docs-ui/**'
   - 'src/sentry/static/sentry/**'
   - 'tests/js/**'
-  - '.github/workflows/javascript*.yml'
+  # - '.github/workflows/js-*.yml'
```
Comment (billyvg) [inline .github/file-filters.yml:15]: TEMP, for testing..

File: .github/workflows/js-getsentry-dispatch.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        uses: actions/github-script@v3
+        if: steps.changes.outputs.frontend == 'true'
+        with:
+          github-token: ${{ steps.getsentry.outputs.token }}
+          script: |
+            github.actions.createWorkflowDispatch({
+              owner: 'getsentry',
+              repo: 'getsentry',
+              workflow_id: 'javascript.yml',
+              ref: 'build/gha/allow-javascript-tests-to-be-skipped',
```
Comment (billyvg) [inline .github/workflows/js-getsentry-dispatch.yml:40]: Also temp, for testing.

File: .github/workflows/js-build-and-lint.yml [hunk 3]
```diff
@@ -1,83 +1,68 @@
...
     branches:
       - master
   pull_request:
-    paths:
-      - '**.[tj]sx?'
-      - package.json
-      - yarn.lock
 
 jobs:
-  getsentry:
```
Comment (billyvg) [inline .github/workflows/js-build-and-lint.yml:14]: Moved this into a new workflow

File: .github/workflows/js-getsentry-dispatch.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          github-token: ${{ steps.getsentry.outputs.token }}
+          script: |
+            github.actions.createWorkflowDispatch({
+              owner: 'getsentry',
+              repo: 'getsentry',
+              workflow_id: 'javascript.yml',
+              ref: 'master',
+              inputs: {
+                'skip': "${{ steps.changes.outputs.frontend != 'true' }}",
```
Comment (joshuarli) [inline .github/workflows/js-getsentry-dispatch.yml:40]:
```suggestion
```

Comment (joshuarli) [reply .github/workflows/js-getsentry-dispatch.yml:40]: Unexpected input.

File: .github/workflows/js-build-and-lint.yml [hunk 5]
```diff
@@ -1,83 +1,68 @@
...
+  ts-and-lint:
```
Comment (joshuarli) [inline .github/workflows/js-build-and-lint.yml:12]: nit: just gonna point out js vs ts inconsistency - probably prefer js everywhere, your choice.

File: .github/workflows/js-build-and-lint.yml [hunk 6]
```diff
@@ -1,83 +1,68 @@
...
 
       - name: eslint + fix
         uses: getsentry/action-eslint-fix@v2
+        if: steps.changes.outputs.frontend == 'true'
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: tsc
-        if: always()
+        if: steps.changes.outputs.frontend == 'true' && always()
```
Comment (joshuarli) [inline .github/workflows/js-build-and-lint.yml:65]:
```suggestion
        if: steps.changes.outputs.frontend == 'true'
```

**Review by joshuarli — APPROVED** [review]
Looks good aside from 1 fix, merge when successful.

File: .github/workflows/js-getsentry-dispatch.yml (hunk 4 above)
Comment (billyvg) [reply .github/workflows/js-getsentry-dispatch.yml:40]: This requires https://github.com/getsentry/getsentry/pull/4622

Comment (joshuarli) [reply .github/workflows/js-getsentry-dispatch.yml:40]: Ohhh.

File: .github/workflows/js-getsentry-dispatch.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# If frontend files change, dispatch a request to getsentry to run its javascript test suites
+name: getsentry dispatcher
+
+on:
+  pull_request:
```
Comment (billyvg) [inline .github/workflows/js-getsentry-dispatch.yml:5]: @joshuarli Going to use `pull_request_target` here too

**Review by joshuarli — APPROVED** [review]

**Pull request merged**